HerPublicWebsite.UnitTests/BusinessLogic/Services/CsvFileCreatorTests.cs

```diff
@@ -90,4 +90,28 @@ public void CreateFileData_CalledWithMultipleReferralRequests_GeneratesExpectedF
 "2023-01-01 13:00:02,Full Name2,contact2@example.com,00002 123456,Address 2 line 1,Address 2 line 2,Town2,County2,AL02 1RS,100 111 222 002,E,yes,Below £31k,no,Owner\r\n" +
 "2023-01-01 13:00:03,Full Name3,contact3@example.com,00003 123456,Address 3 line 1,Address 3 line 2,Town3,County3,AL03 1RS,100 111 222 003,E,yes,Below £31k,no,Owner\r\n");
     }
+
+    [Test]
+    public void CreateFileData_CalledWithSpecialCharacters_GeneratesEscapedFileData()
+    {
+        // Arrange
+        var underTest = new CsvFileCreator();
+        var referralRequest1 = new ReferralRequestBuilder(1).WithFullName("comma,separated,value").Build();
+        var referralRequest2 = new ReferralRequestBuilder(2).WithFullName("double\"quotes").Build();
+        var referralRequest3 = new ReferralRequestBuilder(3).WithFullName("=Formula()").Build();
+        var referralRequest4 = new ReferralRequestBuilder(4).WithTelephone("+441234567890").Build();
+        var referralRequests = new List<ReferralRequest>() { referralRequest1, referralRequest2, referralRequest3, referralRequest4 };
+
+        // Act
+        var data = underTest.CreateFileData(referralRequests);
+
+        // Assert
+        var reader = new StreamReader(data, Encoding.UTF8);
+        reader.ReadToEnd().Should().Be(
+            "Referral date,Name,Email,Telephone,Address1,Address2,Town,County,Postcode,UPRN,EPC Band,Is off gas grid,Household income band,Is eligible postcode,Tenure\r\n" +
+            "2023-01-01 13:00:01,\"comma,separated,value\",contact1@example.com,00001 123456,Address 1 line 1,Address 1 line 2,Town1,County1,AL01 1RS,100 111 222 001,E,yes,Below £31k,no,Owner\r\n" +
+            "2023-01-01 13:00:02,\"double\"\"quotes\",contact2@example.com,00002 123456,Address 2 line 1,Address 2 line 2,Town2,County2,AL02 1RS,100 111 222 002,E,yes,Below £31k,no,Owner\r\n" +
+            "2023-01-01 13:00:03,Formula(),contact3@example.com,00003 123456,Address 3 line 1,Address 3 line 2,Town3,County3,AL03 1RS,100 111 222 003,E,yes,Below £31k,no,Owner\r\n" +
+            "2023-01-01 13:00:04,Full Name4,contact4@example.com,441234567890,Address 4 line 1,Address 4 line 2,Town4,County4,AL04 1RS,100 111 222 004,E,yes,Below £31k,no,Owner\r\n");
```

Comment:
The last case is the one I'm most worried about - it feels a bit wrong to strip the `+` from the phone number, but I'm not sure there's anything better we can do.

It will always render as rubbish in Excel anyway because it'll get parsed as a number :(